src/core/meshcop/border_agent.cpp

```diff
@@ -591,6 +592,35 @@ Error BorderAgent::ForwardToLeader(const Coap::Message &aMessage, const Ip6::Mes
     return error;
 }
 
+void BorderAgent::HandleTmfDatasetGet(Coap::Message          &aMessage,
+                                      const Ip6::MessageInfo &aMessageInfo,
+                                      Dataset::Type           aType)
+{
+    Error          error    = kErrorNone;
+    Coap::Message *response = nullptr;
+
+    if (mState == kStateConnectedAccepted)
+    {
+        Uri uri = (aType == Dataset::kActive) ? kUriActiveGet : kUriPendingGet;
+
+        IgnoreError(ForwardToLeader(aMessage, aMessageInfo, uri));
+        ExitNow();
+    }
+
+    response = (aType == Dataset::kActive) ? Get<ActiveDatasetManager>().ProcessGetRequest(aMessage)
+                                           : Get<PendingDatasetManager>().ProcessGetRequest(aMessage);
+
+    VerifyOrExit(response != nullptr, error = kErrorParse);
+
+    SuccessOrExit(error = Get<Tmf::SecureAgent>().SendMessage(*response, aMessageInfo));
+
+    LogInfo("Sent %sGet response to non-active commissioner", Dataset::TypeToString(aType));
```

Comment:
nit: 
```suggestion
    LogInfo("Sent %s Get response to non-active commissioner", Dataset::TypeToString(aType));
```

Reply:
This is not a typo and intentional.  We use "ActiveGet" and "PendingGet" (with no space) as the URI string.
```
template <> const char *UriToString<kUriActiveGet>(void) { return "ActiveGet"; }
```